src/index/validator.rs

```diff
@@ -0,0 +1,88 @@
+use crate::Dimensions;
+use crate::Embedding;
+use crate::index::actor::AnnError;
+
+pub fn validate_embedding_dimensions(
```

Comment:
As module is called `validator`
```suggestion
pub fn embedding_dimensions(
```